Source/control.cpp

```diff
@@ -1719,9 +1722,35 @@ static int DrawDurIcon4Item(CelOutputBuffer out, ItemStruct *pItem, int x, int c
 			break;
 		}
 	}
-	if (pItem->_iDurability > 2)
-		c += 8;
-	CelDrawTo(out, x, -17 + PANEL_Y, pDurIcons, c, 32);
+
+	int y = -17 + PANEL_Y;
+
+	//Calculate how much of the icon should be gold and red
+	int height = 32;
+	int amount;
+	if (pItem->_iDurability <= DURABILITY_RED)
+		amount = 0;
+	else {
+		if (!sgOptions.Graphics.bDurabilityIconGradual)
+			amount = height;
+		else {
+			int cur = pItem->_iDurability - DURABILITY_RED;
+			amount = (height * cur) / (DURABILITY_GOLD - DURABILITY_RED);
+		}
+	}
+
+	//Draw icon
+	if (amount) {
+		CelOutputBuffer buf = GlobalBackBuffer();
+		buf = buf.subregionY(y - amount, amount);
+		CelDrawTo(buf, x, amount, pDurIcons, c + 8, 32); //Gold icon

```

Comment:
Would be nice to get rid of this `c + 8` in favor of some properly named value. I know it was already like this before, but since we are touching it it could be a good opportunity.